// -----------------------------------------
// 玩家登录
// -----------------------------------------
//查询玩家角色列表
syntax = "proto2";

message CSQueryCharacters {
  //option (msgid) = 1001;
}

// 角色信息
message DCharacter {
  required int64 id           = 1; 		// 角色ID
  optional string name        = 2; 		// 昵称
  optional int32 profession   = 3; 		// 职业
  optional int32 level        = 4; 		// 等级
  optional int32 sex          = 5; 		// 性别
  optional string model       = 6; 		// 选择的模型
  optional int64 survive_time = 7; 		// 剩余删除时间，单位(秒) 0 = 永久
  optional int32 equipWeapon1 = 8; 		// 主武器
  optional int32 equipWeapon2 = 9; 		// 副武器
  optional int32 vipLevel     = 10;		// VIP等级
  optional int32 sn           = 11;		// 角色表sn
}

//返回玩家角色列表
message SCQueryCharactersResult {
  //option (msgid) = 1002;
  repeated DCharacter characters = 1; //角色列表集合 支持多角色
}

//创建角色
message CSCharacterCreate {
  //option (msgid) = 1003;
  required string name = 1; //昵称
  optional string keyActivate = 2; //激活码
}

//创建角色返回
message SCCharacterCreateResult {
  //option (msgid) = 1004;
  required int32 resultCode = 1; //执行结果编码
  optional string resultReason = 2; //执行结果描述
  optional int64 humanId = 3; //新创建角色ID
  optional string url = 4; //url跳转（resultCode = -2）
}

//删除角色
message CSCharacterDelete {
  //option (msgid) = 1005;
  required int64 id = 1; //角色ID
}

//删除角色结果
message SCCharacterDeleteResult {
  //option (msgid) = 1006;
  required int32 resultCode = 1; //执行结果编码
  optional int64 id = 2;
  optional int64 survive_time = 3; //删除成功后,该角色的剩余删除时间,-1时直接从列表中移除
}

//角色登录
message CSCharacterLogin {
  //option (msgid) = 1007;
  required int64 humanId = 1; //角色ID
}

enum ELoginType{
  LOGIN_DEFAULT   = 0;  // 未使用
  DAY_FIRST_LOGIN = 1;	// 当天第一次登陆
  RECOVER_LOGIN   = 2;	// 未真正下线，服务器缓存
  DAY_MULTI_LOGIN = 3;  // 当天多次登陆
}

//角色登录结果
message SCCharacterLoginResult {
  //option (msgid) = 1008;
  required int32 resultCode = 1; //执行结果编码
  optional ELoginType loginType = 2; // 本次登陆类型
}

// 取消删除
message CSCharacterCancelDelete {
  //option (msgid) = 1009;
  required int64 id = 1; //角色id
}

// 取消删除返回
message SCCharacterCancelDeleteResult {
  //option (msgid) = 1010;
  required int32 resultCode = 1; //错误码0=成功
  optional int64 id = 2; //角色id
}

//退出登录
message CSCharacterExit {
  //option (msgid) = 1011;
}